### PR TITLE
Show correct URL when using retrolab

### PR DESCRIPTION
**Pull request by yuvipanda**
- Provides direct link to notebook / directory the user is
  currently viewing
- Adds a JS dpeendency on @retrolab/application - I *think* this
  is the right way to detect if we are in retrolab? Alternatively
  we could check for app.name or similar. I'm hopefull this does
  not actually increase the built size of the extension too much?

Fixes https://github.com/jupyterlab-contrib/jupyterlab-link-share/issues/25

### CODE REVIEW

**Comment by yuvipanda** [comment]
Should I not commit yarn.lock, @jtpio?

**Comment by jtpio** [comment]
There is a new dependency so the changes in `yarn.lock` are expected.

You can run `jlpm` (or `yarn`) locally and the conflicts should be fixed automatically (yarn does it for you)

**Comment by jtpio** [comment]
> I think this is the right way to detect if we are in retrolab?

For now yes. I think the LSP extension uses a similar approach.

**Comment by yuvipanda** [comment]
@jtpio great! done.

File: src/index.ts [hunk 1]
```diff
@@ -58,11 +61,20 @@ const plugin: JupyterFrontEndPlugin<void> = {
         }
 
         const links = results.map(server => {
-          return URLExt.normalize(
-            `${PageConfig.getUrl({
-              workspace: PageConfig.defaultWorkspace
-            })}?token=${server.token}`
-          );
+          if (retroShell !== null) {
```
Comment (jtpio) [inline src/index.ts:64]: Often we use the more canonical form (mostly cosmetics / readability):

```suggestion
          if (retroShell) {
```

Comment (yuvipanda) [reply src/index.ts:64]: aaah, makes sense given this is typescript so it can't be something like `false` or `0`

Comment (yuvipanda) [reply src/index.ts]: Fixed

**Comment by jtpio** [comment]
Thanks!

Maybe we could also add a `.binder` folder to install `retrolab` so it can easily be tested on Binder?

We would also need a `postBuild` that looks similar to this one: https://github.com/jupyterlab/retrolab/blob/main/binder/postBuild

**Review by jtpio — APPROVED** [review]
Thanks!

**Comment by jtpio** [comment]
> Maybe we could also add a .binder folder to install retrolab so it can easily be tested on Binder?

We can do that in a separate PR.

**Comment by yuvipanda** [comment]
@jtpio great! Think you can do a release this time too? :D

**Comment by jtpio** [comment]
Sure. Unless you would like to give it a try with the releaser (requires a bit of (one time) setup though).

**Comment by yuvipanda** [comment]
@jtpio i'll do that next week for sure!

**Comment by jtpio** [comment]
Nice, thanks!

Releasing now.

**Comment by jtpio** [comment]
@yuvipanda https://github.com/jupyterlab-contrib/jupyterlab-link-share/releases/tag/v0.2.4

**Comment by yuvipanda** [comment]
@jtpio great, deployed!

**Pull request merged**